drivers/android/transaction.rs

```diff
@@ -136,6 +169,34 @@ impl Transaction {
             process.push_work(self)
         }
     }
+
+    /// Prepares the file list for delivery to the caller.
+    fn prepare_file_list(&self) -> Result<List<Box<FileInfo>>> {
+        // Get list of files that are being transfered as part of the transaction.
+        let mut file_list = List::new();
+        core::mem::swap(&mut file_list, &mut self.inner.lock().file_list);
```

Comment:
You can use `mem::replace` here too. `let mut file_list = core::mem::replace(&mut self.inner.lock().file_list, List::new());`

Reply:
Will change this too.